src/main/java/com/squareup/javapoet/TypeVariableName.java

```diff
@@ -57,6 +57,18 @@ private TypeVariableName(String name, List<TypeName> bounds, List<AnnotationSpec
     return new TypeVariableName(name, bounds);
   }
 
+  public TypeVariableName withBounds(Type... bounds) {
+    return get(name, bounds);
```

Comment:
Append, but I don't think it matters. They should probably be a Set in actuality.